packages/launcher/src/renderer/launcher/contacts/InviteContactModal.js

```diff
@@ -0,0 +1,434 @@
+//@flow
+import React, { Component } from 'react'
+import styled from 'styled-components/native'
+import { Text } from '@morpheus-ui/core'
+import { graphql, createFragmentContainer } from 'react-relay'
+import Loader from '../../UIComponents/Loader'
+
+import rpc from '../rpc'
+import FormModalView from '../../UIComponents/FormModalView'
+import { EnvironmentContext } from '../RelayEnvironment'
+import applyContext, { type ContextProps } from '../LauncherContext'
+
+import type { InviteContactModal_contact as Contact } from './__generated__/InviteContactModal_contact.graphql.js'
+
+export type TransactionType = 'invite' | 'retrieveStake' | 'declineInvite'
+
+type Props = ContextProps & {
+  closeModal: () => void,
+  contact: Contact,
+  type: TransactionType,
+}
+
+type TXParams = {
+  gasPriceGwei: string,
+  maxCost: string,
+  stakeAmount: string,
+}
+
+type State = {
+  txProcessing: boolean,
+  error?: ?string,
+  txParams?: ?TXParams,
+  declinedTXHash?: ?string,
+  withdrawTXHash?: ?string,
+  balances?: {
+    eth: string,
+    mft: string,
+  },
+  invitePending?: {
+    error?: string,
+    state:
+      | 'awaiting_approval'
+      | 'sending_approval'
+      | 'approved'
+      | 'sending_invite'
+      | 'invite_sent'
+      | 'error',
+  },
+}
+
+const MFT_TOKEN_ADDRESSES = {
+  ropsten: '0xa46f1563984209fe47f8236f8b01a03f03f957e4',
+  mainnet: '0xdf2c7238198ad8b389666574f2d8bc411a4b7428',
+}
+
+const FormContainer = styled.View`
+  margin-top: 20px;
+  max-width: 450px;
+  flex: 1;
+  width: 100%;
+  align-self: center;
+  align-items: center;
+  margin-top: 15vh;
+`
+
+const Section = styled.View`
+  margin-bottom: 20px;
+`
+
+export class InviteContactModal extends Component<Props, State> {
+  static contextType = EnvironmentContext
```

Comment:
Relay's `createFragmentContainer()` injects the [`relay` prop](https://facebook.github.io/relay/docs/en/v1.7.0/fragment-container.html#available-props) in the wrapped component so it shouldn't be needed to use the context.